sw/airborne/modules/nav/nav_survey_poly_rotorcraft.c

```diff
@@ -155,7 +155,7 @@ bool nav_survey_poly_setup(uint8_t EntryWP, uint8_t Size, float sw, float Orient
   CSurveyStatus = Init;
 
   if (Size == 0) {
-    return true;
+    return;
```

Comment:
The OK as for rotorcraft, quick glance fixedwing look OK, Thx. Note that The call_once (or call for that matter ) was discussed quite some time ago. For now to go forward i think it is fine. In future I rather see a nice non tech flightplan descriptive language and users writing flightplans should just be able to write e.g simply <SetGroundReferenceHere"/>. But that is another discussion. Yes I know, one could write a wrapper for nice flighplan language encapsulating current implementation.

BTW as for SurveySize returning if 0, well if I did not arrive at my survey area one could wish to set it as moving twarth gathering data or let it tepend on fuel reserve. now it cannot move toward But if ever needed it that way myself, I'll file a new issue ;)

BTW not testflown and simmed yet, will do, with finger on TX Mode switch ;) when it lnds in master and hopefully not -20 deg C.